### PR TITLE
Improve implementation of the `NetworkError` enum

### LINKED ISSUE #9: Improve implementation of the `NetworkError` enum
The current task is part of #6

We have [implemented the `NetworkError` enum](#5) to describe possible app-specific errors in the network layer.
Several improvements are still required:
- We have to improve conformance to the `allCases` to auto-generate possible cases within tests to make tests more robust;
- Additional error `badResponse` should be introduced to handle the case when the HTTP response is invalid;
- (Optionally) tests can be simplified a bit.

### PR DESCRIPTION
**Pull request by dersim-davaod**
Resolves #9

### CODE REVIEW

File: Sources/NetworkService/NetworkError/NetworkError.swift [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 
 /// Contains network errors that may occur, while the application is running.
-public enum NetworkError: Error {
+public enum NetworkError: Error, CaseIterable {
     /// Indicates that the server cannot process the request due to a client error
     /// (e.g., malformed request syntax, size too large, invalid request message framing,
     /// or deceptive request routing).
     case badRequest
 
+    /// Indicates that the response from the serveris invalid or the client couldn't parse it.
```
Comment (YuliaNastiushkina) [inline Sources/NetworkService/NetworkError/NetworkError.swift]: Typo ?

```suggestion
    /// Indicates that the response from the server is invalid or the client couldn't parse it.
```

Comment (marinkman) [inline Sources/NetworkService/NetworkError/NetworkError.swift]:
```suggestion
    /// Indicates that the response from the server is invalid or the client couldn't parse it.
```

File: Tests/NetworkServiceTests/NetworkErrorTests/NetworkError+LocalizedErrorTests.swift [hunk 2]
```diff
@@ -2,60 +2,26 @@ import NetworkService
...
-        XCTAssertEqual(expectedLocalizedDescription, actualLocalizedDescription)
-    }
-
-    func testLocalizedDescriptionReturnsCorrespondingMessageForUnknownError() {
-        // Given
-        let expectedLocalizedDescription = "Unknown error."
-        let actualLocalizedDescription = NetworkError.unknownError.localizedDescription
+        let clientErrorDescription = "The server cannot process the request due to a client error."
+        let serverErrorDescription = "The server cannot process the request due to a issue with the server."
+        let requestTimeoutDescription = "The server timed out waiting for the request." // TODO: response?
```
Comment (YuliaNastiushkina) [inline Tests/NetworkServiceTests/NetworkErrorTests/NetworkError+LocalizedErrorTests.swift]: Why do we need TODO here?

Comment (marinkman) [inline Tests/NetworkServiceTests/NetworkErrorTests/NetworkError+LocalizedErrorTests.swift]: `// TODO: response?`

Why is it here?

**Review by YuliaNastiushkina — COMMENTED** [review]
Nice job!
I have some questions, check it, please.

**Review by iBYarchitect — APPROVED** [review]
LGTM

**Review by kirpichonok — COMMENTED** [review]
LGTM, after fixing comments of the reviewers.

File: Sources/NetworkService/NetworkError/NetworkError.swift (hunk 1 above)
Comment (dersim-davaod) [reply Sources/NetworkService/NetworkError/NetworkError.swift]: Right, thanks!

File: Tests/NetworkServiceTests/NetworkErrorTests/NetworkError+LocalizedErrorTests.swift (hunk 2 above)
Comment (dersim-davaod) [reply Tests/NetworkServiceTests/NetworkErrorTests/NetworkError+LocalizedErrorTests.swift]: Forgot to update the wrong error's description. Good catch!

File: Sources/NetworkService/NetworkError/NetworkError.swift (hunk 1 above)
Comment (dersim-davaod) [reply Sources/NetworkService/NetworkError/NetworkError.swift]: Fixed.

**Review by YuliaNastiushkina — APPROVED** [review]
LGTM

**Review by marinkman — COMMENTED** [review]
Please make changes to the `CHANGELOG.md` file.

**Review by iBYarchitect — APPROVED** [review]
LGTM

**Review by NadzeyaShpakouskaya — APPROVED** [review]
LGTM

**Review by kirpichonok — APPROVED** [review]
LGTM

**Pull request merged**